### ISSUE TITLE
Running prediction on GPU

### ISSUE THREAD

**Issue by theycallhimavi** [catboost/catboost#560]
Problem: Is it possible to run catboost predict on GPU
catboost version: latest
Operating System: linux
CPU: 16 core Xeon
GPU: 8x Tesla V100s

**Comment by Noxoomo** [comment]
We don't have GPU predictor yet, only learning, metric computations and predictions computation on test set.

For some applications Catboost could benefit from GPU predictor (e.g. many decision trees), but we have no resources to implement in the near future.

**Comment by theycallhimavi** [comment]
Is it possible to use it by exporting to some other format? If not, how can i increase the speed of prediction?

**Comment by Noxoomo** [comment]
Could you please specify properties of you dataset (number of float and categorical features, documents), number of trees, tree depth

And time it takes to apply catboost to you model?
Do you use FeaturesData class to speed-up apply? 
Do you just apply model, or doing stage prediction?

**Comment by theycallhimavi** [comment]
My dataset currently has 3 float features and 10 categorical features. Im trying to get to use it for Reat time bidding Ad RPM prediction. The training dataset varies from 1 billion to 100 million records and the  prediction dataset is around 50 million records.
Tree Depth - 10
Number of trees - 10k
I dont know how FeaturesData class can speed things up
I just apply the model. Im not using stage prediction

**Comment by annaveronika** [comment]
Features data will help, because if you just use list of lists then data will be copied all around. So try it first.
Are your categorical features strings or numbers? If they are numbers then we will probably be able to speed up the prediction by a lot by allowing to pass it as numbers and by not calculating hashes on them. We do have it planned.

**Comment by DanielLumb** [comment]
Came here to ask about GPU prediction in catboost.
Release notes for 0.16 said - to my amazement and joy - that "It is now possible to apply model on GPU"
I looked around Python API & documentation quite hard, and cannot find any clue how to use model evaluation/prediction working on GPU.

Can anyone give a hint about how to use that very promising new feature?
Thank you!

**Comment by annaveronika** [comment]
This is implemented now.
